Feature/protected routes

JatsukinG opened the pull request:
Created new components and Custom hook for login state, so that different routes can be protected if the user is not logged in.

JatsukinG committed:
Added logic for redirect to dashboard if user is login

JatsukinG committed:
Created localStorage variable to save authentication token

JatsukinG committed:
Created Authentication context

JatsukinG committed:
Added state for AuthContext

JatsukinG committed:
Create useAuth hook for user authentication

improved code in Login form and redirect component

JatsukinG committed:
Create Router component to protect routes according user login state

cristiangrojas commented on src/components/auth/LoginForm/index.jsx:
```diff
@@ -1,38 +1,42 @@
-    const [userInputValue,setUserInputValue] = useState('')
-    const [passwordInputValue,setPasswordInputValue] = useState('')
+    const {login} = useAuth()
+    const [userInputValue, setUserInputValue] = useState('jatmolina15@gmail.com')
```
remove default values from username and password, keep just for development purposes

cristiangrojas commented on src/hooks/auth/useAuth.js:
```diff
@@ -0,0 +1,28 @@
+import {useContext} from "react";
+import {AuthContext} from "providers/AuthProvider";
+
+export const LOCAL_STORAGE_AUTH_TOKEN_KEY = 'token'
```
move this constant into `/src/constants/localStorage.js`

JatsukinG committed:
Removed default inputs values

Moved constant export const LOCAL_STORAGE_AUTH_TOKEN_KEY into "src/constants/localStorage.js"

JatsukinG committed:
solved conflicts to log in and protect routes

Added message error below password input

cristiangrojas commented on src/components/auth/LoginForm/index.jsx:
```diff
@@ -1,38 +1,51 @@
+                body: JSON.stringify(body),
+            })
+            const data = await response.json()
+            console.log(data);
```
remove console.log

cristiangrojas commented on src/components/auth/LoginForm/index.jsx:
```diff
@@ -1,38 +1,51 @@
+            if(data.error){
+                setShowErrorMessage(true)
+                setErrorMessage(data.error.message)
+
```
remove extra space

cristiangrojas commented on src/components/auth/LoginForm/index.jsx:
```diff
@@ -1,38 +1,51 @@
+
+            } else
+                login(data.jwt)
+
```
too many extra white spaces

cristiangrojas commented on src/components/auth/LoginForm/index.jsx:
```diff
@@ -52,6 +65,9 @@ export const LoginForm = () => {
                             placeholder="Contraseña"/>
                     </div>
+                    {
+                        showErrorMessage && <p className="text-red-500 text-sm font-medium mb-4">{errorMessage}</p>
```
If you use something more than once in different components, creating a new reusable component may be a good idea

In this case let's make an `ErrorMessage` component in `/src/components/forms/ErrorMessage`

Keep in mind margin at bottom shouldn't be a default

JatsukinG committed:
Create component to return error message

improved code organization

JatsukinG committed:
Used error message component in searching plate form

JatsukinG committed:
Merge branch 'release/1.0.0' into feature/protected_routes
